helios-services/src/main/java/com/spotify/helios/agent/AgentParser.java

```diff
@@ -149,25 +152,37 @@ public AgentParser(final String... args) throws ArgumentParserException {
       agentConfig.setId(base16().encode(idBytes));
     }
 
-    final List<String> dns = options.getList(dnsArg.getDest());
-    if (!dns.isEmpty()) {
-      for (final String d : dns) {
-        if (!isInetAddress(d)) {
-          throw new IllegalArgumentException("Invalid IP address " + d);
-        }
-      }
-    }
-    agentConfig.setDns(dns);
-
-    final List<String> binds = options.getList(bindArg.getDest());
-    if (!binds.isEmpty()) {
-      for (final String b : binds) {
-        if (!isValidBind(b)) {
-          throw new IllegalArgumentException("Invalid bind " + b);
-        }
-      }
+    agentConfig.setDns(validateArgument(
+        options.getList(dnsArg.getDest()),
+        InetAddresses::isInetAddress,
+        arg -> "Invalid IP address " + arg));
+
+    agentConfig.setBinds(validateArgument(
+        options.getList(bindArg.getDest()),
+        BindVolumeContainerDecorator::isValidBind,
+        arg -> "Invalid bind " + arg));
+
+    agentConfig.setExtraHosts(validateArgument(
+        options.getList(addHostArg.getDest()),
+        AddExtraHostContainerDecorator::isValidArg,
+        arg -> "Invalid ExtraHost " + arg));
```

Comment:
please tell me if this and the `validateArgument` function below is way too much functional-style